### PR TITLE
Produce AttachEvent to Kafka on Envoy attachment

### PR DESCRIPTION
**Pull request by itzg**
# Resolves

SALUS-136

# What

Produces an AttachEvent to Kafka/UMB during the Envoy registration process.

# How

Adds to the completion chain that processes the Envoy registration/attachment. It uses the JsonSerializer to simplify the KafkaTemplate call.

## How to test

A unit test was added that focuses just on the postEvent operation. The other operations are completely mocked out and not verified.

# Why

n/a

# TODO

I did some manual testing of an Envoy attaching, but I'd like to verify the message was actually sent with the expected JSON payload.

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -5,11 +5,17 @@ ambassador:
...
+      telemetry.event.resource.json
 etcd:
   url: http://localhost:2479
 spring:
   resources:
+    # disable serving of static web files since this is a REST/Actuator only web app
     add-mappings: false
+  kafka:
+    producer:
+      value-serializer: org.springframework.kafka.support.serializer.JsonSerializer
```
Comment (jjbuchan) [inline src/main/resources/application.yml]: I've had to back out of using this.  The consumer is unable to convert the AttachEvent string/json back to the actual object.

This fails

```
public Object fromMessage(Message<?> message, Class<?> targetClass) {
    Object payload = message.getPayload();
    if (this.conversionService.canConvert(payload.getClass(), targetClass)) {
        try {
            return this.conversionService.convert(payload, targetClass);
        } catch (ConversionException var5) {
            throw new MessageConversionException(message, "Failed to convert message payload '" + payload + "' to '" + targetClass.getName() + "'", var5);
        }
    } else {
        return ClassUtils.isAssignableValue(targetClass, payload) ? payload : null;
    }
}
```

Comment (itzg) [reply src/main/resources/application.yml]: That’s strange. I’ll do some more testing of the produced message to see if I can spot something wrong on this end.

Comment (itzg) [reply src/main/resources/application.yml]: @jjbuchan FYI here's what one of the attach events looks like for me. I'm using my UMB consumer so I can see the headers along with message key and payload:

```
% Message on telemetry.event.resource.json[0]@2 (key=aaaaaa:hostname:MS90HCG8WL):
% Headers:{__TypeId__=com.rackspace.salus.telemetry.events.AttachEvent}
{"identifierName":"hostname","identifierValue":"MS90HCG8WL","labels":{"os":"DARWIN","hostname":"MS90HCG8WL","arch":"X86_64"},"envoyId":"aeafafa4-18e0-11e9-b2e7-6a00027f65d0","tenantId":"aaaaaa","envoyAddress":"0:0:0:0:0:0:0:1"}
```

That `__TypeId__` is not the header name I remember seeing, so they might have renamed recently. I wonder if we're using a mismatch of Spring Kafka versions between apps.

Comment (jjbuchan) [reply src/main/resources/application.yml]: For whatever reason, this does seem to work for me now.

**Pull request merged**